File: magellan-common/src/main/java/com/wealthfront/magellan/NavigationType.java
package com.wealthfront.magellan;

/**
 * Represents the different types of navigation we support:
 * <ul>
 *   <li>SHOW: represent a modal type of Navigation, where the Screen is only shown if not already displayed, and
 *   with a default animation of sliding the new screen from the bottom.</li>
 *   <li>GO: is the normal type of navigation, with a default animation of sliding both screens in sync from the right
 *   to the left.</li>
 *   <li>NO_ANIM: represent a normal navigation but with no animation.</li>
 * </ul>
 */
public enum NavigationType {
  SHOW,
  GO,
  NO_ANIM
}

File: magellan-kotlin/src/main/java/com/wealthfront/magellan/kotlin/NavigationType.kt
package com.wealthfront.magellan.kotlin

/**
 * Represents the different types of navigation we support:
 *
 *  * SHOW: represent a modal type of Navigation, where the Screen is only shown if not already displayed, and
 * with a default animation of sliding the new screen from the bottom.
 *  * GO: is the normal type of navigation, with a default animation of sliding both screens in sync from the right
 * to the left.
 *  * NO_ANIM: represent a normal navigation but with no animation.
 *
 */
enum class NavigationType {
  SHOW,
  GO,
  NO_ANIM
}
